src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -192,17 +192,21 @@ final case class EServicesApiServiceImpl(
 
   private def getProducerEServicesIds(producerId: UUID, consumersIds: List[String])(implicit
     contexts: Seq[(String, String)]
-  ): Future[List[UUID]] =
-    Future
-      .traverse(consumersIds)(consumerId =>
-        // TODO This call will be paginated
-        agreementProcessService.getAgreements(
-          producerId = producerId.toString.some,
-          consumerId.some,
-          states = List(AgreementProcess.AgreementState.ACTIVE, AgreementProcess.AgreementState.SUSPENDED)
+  ): Future[List[UUID]] = {
+    for {
+      agreements <- Future
+        .traverse(consumersIds)(consumerId =>
+          agreementProcessService.getAgreements(
+            consumersIds = Seq(UUID.fromString(consumerId)),
```

Comment:
`UUID.fromString` raises an exception if the string is not an UUID.
It's preferable to convert the `consumersIds` before passing it to this function.
You can use the `toFutureUUID` helper